src/dmd/link.d

```diff
@@ -34,6 +35,23 @@ version (Posix) extern (C) int pipe(int*);
 version (Windows) extern (C) int spawnlp(int, const char*, const char*, const char*, const char*);
 version (Windows) extern (C) int spawnl(int, const char*, const char*, const char*, const char*);
 version (Windows) extern (C) int spawnv(int, const char*, const char**);
+version (Posix) extern (C) extern __gshared char** environ;
+version (CRuntime_Glibc)
+{
+    // borrowed from https://github.com/dlang/druntime/commit/51a9962482a668fab00cc866045d76a9c4a39b63
+    // remove and replace with core.sys.posix.spawn once 2.083 is bootstrap
+    extern(C):
+    struct posix_spawn_file_actions_t { void[80] pad; } // idk what is even in here but the sizeof matches
```

Comment:
Having an opaque `void[80]` here is definitely *not* an option.
Especially since you linked to the commit with the correct definition, why can't it just be copied over ?
A rapid inspection of the struct shows that [it contains a pointer](https://github.com/dlang/druntime/commit/51a9962482a668fab00cc866045d76a9c4a39b63#diff-767e473e72d6e23aca965ce3ca408b4aR110-R116), so that value is wrong on 32 bits.

Reply:
Right, now that we're on version glibc I can just copy it. I didn't want to depend on internals pointlessly, but you're right about the pointer - not that it matters, worst case we'd just use 4 bytes extra.

Reply:
> edit: The problem is we have no way of depending on the glibc data type layout. So it might as well be opaque - if glibc changes its type size, how would we know anyways? There's not even a size parameter on the function. This whole API is not very satisfying.

I know it is all hypothetical, because C developers care about binary compatibility so it's extremely unlikely it happens, but if it does, will it be easier to spot if we have the expected fields, and can compare it to the C headers on our machine, or just a `void[80]` blob ?

Reply:
I honestly feel the void[80] blob is easier to compare, because it doesn't imply we're going by some official specification, and also you only need to check `sizeof` on the C side. It implies the `sizeof` is the important part, not the fields. 

I'll change it on monday though, I don't really care strongly about this.

Reply:
done